packages/jaeger-ui/src/components/TracePage/CriticalPath/testResults/test1.js

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import transformTraceData from '../../../../model/transform-trace-data';
+
+/*
```

Comment:
a few general comments on this:
  * unit tests should be simple, validating individual use cases, like "child end after parent". Such use case requires a test trace with just two spans. I am seeing much more complex test traces here, why is that?
  * test data should be easily understandable. 
    * When I look at the first span in happyTrace, it is 17 lines of code. Is all that complexity required to test a use case of the algorithm? Or could we achieve the same with something like `createChildSpan("childA", "parentA", 1, 10)`?
     * when I look at `test1.js`, it's >100 lines, using UUIDs for span IDs - this is not easy to understand
     * `const startTime = 1579070675083000` - why not 0? This long number just adds cognitive complexity, for no apparent benefit.
  * ideally there should be clear separation between executable code of the unit test, and the data fixtures. I am not yet convinced that data fixtures are even necessary here, as they make the tests harder to understand. Say we want to unit test a case of parent with two parallel children. The critical path structure would have 2-4 segments at most. It's much easier to just encode those directly in the test than to look in 3 different files (test, input, & output) to understand the scenario.

Reply:
Yepp, Now I trimmed the data upto 3 spans atmost and I also used normal numbers (in place of UUIDs ). Let me know if there are any more suggestions or modifications to do